lib/modules/apostrophe-widgets/public/js/user.js

```diff
@@ -8,17 +8,33 @@ apos.define('apostrophe-widgets', {
 
     // Opens the editor modal of a widget, unless the widget is contextualOnly,
     // in which case we simply save the widget and call the save method
+    // Widget can opitonally be set to skipInitialModal which skips the first
+    // edit modal but binds future editing interactions
+
     self.edit = function(data, options, save) {
-      // Only create a modal editor if the schema is not contextual only
-      if(self.options.contextualOnly){
-        // Check for an id or assign a generated one
+      
+      // Check for an id or assign a generated one
+      var checkOrGenerateId = function(data) {
```

Comment:
Please make this an ordinary named function (not an assignment statement) and move it to the end of the function so it doesn't interrupt easy "top down" reading of what the method does.